src/librustc_mir/interpret/traits.rs

```diff
@@ -83,7 +83,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                     self.param_env,
                     def_id,
                     substs,
-                ).ok_or_else(|| InterpError::TooGeneric)?;
+                ).ok_or_else(|| InterpError::InvalidProgram(InvalidProgramInfo::TooGeneric))?;
```

Comment:
I feel like we want a macro for these "map" cases as well. It would be like `err` but not wrap the `Err` around and not call `into()`.

`error!` is already taken by the log crate, I think. @oli-obk any suggestions for how to call that macro?

Reply:
@RalfJung are you suggesting a macro only for the 
`|| InterpError::InvalidProgram(InvalidProgramInfo::TooGeneric)` part?

Reply:
Likely without the `||` even, yes. I think that's as much as we can macro-ize (while maintaining postfix position).

Reply:
Ah, so you mean something like this: `.ok_or_else(|| err_inval!(TooGeneric))?`?

Reply:
Yes, exactly.

Reply:
A macro starting with `err` would not be appropriate here right? We just get the fully qualified name for the enum variant here. And again we would need to have a macro for each `Info` variant from `InterpError`.

Reply:
Yes. If we could add it to the `InterpError` namespace, that would be fun, but I doubt that's possible (like, `InterpError::inval!(...)`). We can always do `interp::error::inval!` but that's too many `::` for my taste.

Reply:
You can always invent an `interp_error` module that has `inval` and other macros

Reply:
@RalfJung @oli-obk I am going to create multiple macros to serve this purpose. Hope that's fine.

Reply:
I had another idea:
* rename the existing macros to `throw_*` (e.g., `throw_ub!`), and make them `return Err(...into())`.
* then have new macros `err_*` that just create the `InterpError` variant.

Reply:
Makes sense. But should they be prefixed with `err_*` when actually we are just creating an `InterpError` variant?

Reply:
`error_` seems fine, too. They do create an `InterpError`, after all.

Reply:
Done.

Reply:
Yeah, something like that -- I left some comments in a new conversation though, mostly asking about why you deviated from my proposal in two ways. ;)

@oli-obk does this seem reasonable to you?